src/app/devices/deviceContent/components/cloudToDeviceMessage/cloudToDeviceMessage.tsx

```diff
@@ -0,0 +1,324 @@
+                        checkButtonAriaLabel={context.t(ResourceKeys.cloudToDeviceMessage.properties.rowCheckBoxAriaLabel)}
+                        selection={this.state.selection}
+                    />
+                </MarqueeSelection>
+            </>
+        );
+    }
+
+    private readonly onSelectionChanged = () => {
+        this.setState({
+            selectedIndices: new Set(this.state.selection.getSelectedIndices())
+        });
+    }
+
+    private readonly handleAddCustomProperty = () => {
+        this.setState(prevState => ({
+            properties: [...prevState.properties, {isSystemProperty: false, index: prevState.properties.length + 1, keyName: '', value: ''}]
+        }));
+    }
+
+    private readonly handleAddSystemProperty = (keyName: string) => () => {
+        this.setState(prevState => ({
+            properties: [...prevState.properties, {isSystemProperty: true,  index: prevState.properties.length + 1, keyName, value: ''}]
+        }));
+    }
+
+    private readonly handleEditCustomPropertyKey = (property: PropertyItem) => ((event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+        const items = this.state.properties;
+        items.forEach((element, index) => {
+            if (element.index === property.index) {
+                items[index] = {...items[index], keyName: newValue};
+            }
+        });
+        this.setState({properties: items});
+    })
+
+    private readonly handleEditPropertyValue = (property: PropertyItem) => ((event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+        const items = this.state.properties;
+        items.forEach((element, index) => {
+            if (element.index === property.index) {
+                items[index] = {...items[index], value: newValue};
+            }
+        });
+        this.setState({properties: items});
+    })
+
+    private readonly handleDelete = () => {
+        const updatedProperties = [];
+        for (let i = 0; i < this.state.properties.length; i++) {
+            if (!this.state.selectedIndices.has(i)) {
+                updatedProperties.push(this.state.properties[i]);
+            }
+        }
+        this.setState({properties: updatedProperties});
+    }
+
+    private readonly onTextFieldChange = (ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newText: string) => {
+        this.setState({
+            body: newText
+        });
+    }
+
+    private readonly onCheckboxChange = (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: boolean) => {
+        this.setState({
+            addTimestamp: checked
+        });
+    }
+
+    private readonly toggleCollapse = () => {
+        this.setState({
+            propertiesSectionCollapsed: !this.state.propertiesSectionCollapsed
+        });
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    private readonly onSendMessageClick = () => {
```

Comment:
This method can get some simplification and remove the cyclomatic complexity disabling.